packages/transformer/src/IModelTransformer.ts

```diff
@@ -2889,145 +2951,227 @@ export class IModelTransformer extends IModelExportHandler {
+    sourceIdOfRelationshipInSource: Id64String,
+    targetIdOfRelationshipInSource: Id64String
+  ) {
+    // we need a connected iModel with changes to remap elements with deletions
+    const notConnectedModel = this.sourceDb.iTwinId === undefined;
+    const noChanges =
+      this.synchronizationVersion.index === this.sourceDb.changeset.index &&
+      (this.exporter.sourceDbChanges === undefined ||
+        !this.exporter.sourceDbChanges.hasChanges);
+    if (notConnectedModel || noChanges) return;
+
+    const sourceIdOfRelationshipInTarget = await this.getTargetIdFromSourceId(
+      sourceIdOfRelationshipInSource,
+      true
+    );
+    const targetIdOfRelationshipInTarget = await this.getTargetIdFromSourceId(
+      targetIdOfRelationshipInSource,
+      true
+    );
+    if (sourceIdOfRelationshipInTarget && targetIdOfRelationshipInTarget) {
+      this._deletedSourceRelationshipData!.set(changedInstanceId, {
+        classFullName,
+        sourceIdInTarget: sourceIdOfRelationshipInTarget,
+        targetIdInTarget: targetIdOfRelationshipInTarget,
+      });
+    } else if (this.sourceDb === this.provenanceSourceDb) {
+      const relProvenance = this._queryProvenanceForRelationship(
+        changedInstanceId,
+        {
+          classFullName,
+          sourceId: sourceIdOfRelationshipInSource,
+          targetId: targetIdOfRelationshipInSource,
+        }
+      );
+      if (relProvenance && relProvenance.relationshipId)
+        this._deletedSourceRelationshipData!.set(changedInstanceId, {
+          classFullName,
+          relId: relProvenance.relationshipId,
+          provenanceAspectId: relProvenance.aspectId,
+        });
+    }
+  }
+
+  /**
+   * This function is called by the transformer as it is about to process the changesets passed to it in [[IModelTransformOptions.argsForProcessChanges]].
+   * This would be after the exporter has already processed the same set of changesets passed to the transformer in [[IModelTransformOptions.argsForProcessChanges]].
+   * This function should be used to modify the exporter's sourceDbChanges, if necessary, using [[ChangedInstanceIds.addCustomChange]]. See [[ChangedInstanceIds.addCustomChange]] for more information.
+   * @param sourceDbChanges will only be defined if the transformer was called with [[IModelTransformOptions.argsForProcessChanges]].
+   * @note If defined, sourceDbChanges will already be populated with the changesets passed to the transformer, if any when this function is called by the transformer.
+   * @note The transformer will have built up the remap table between the source and target iModels before calling this function. This means that functions like [[IModelTransformer.context.findTargetElementId]] will return meaningful results.
+   * @note Its expected that this function be overridden by a subclass of transformer if it needs to modify sourceDbChanges.
+   */
+  protected async addCustomChanges(
+    _sourceDbChanges?: ChangedInstanceIds
```

Comment:
`_sourceDbChanges?` should be required. As API user I have no idea how to handle undefined source db changes. It is impossible to add custom changes and I am too worried to throw error (what if it is expected). 
It should be transformers responsibility to handle case when source db change is undefined and act accordingly (throw / not to call addCustomChanges / initalize custom changes / something else).

Reply:
made sourceDbChanges required, and only call addCustomChanges if its defined.